### ISSUE TITLE
Allow initializing class instances without `.new()`

### ISSUE THREAD

**Issue by filantus** [godotengine/godot-proposals#4707]
### Describe the project you are working on

General GDScript improvements.

### Describe the problem or limitation you are having in your project

In a matter of syntax shugar, is there any chance to allow to initialize custom GDScript classes without `.new()`?

For example:
```gdscript
var value: CustomCollection = CustomCollection()
```

Just like `Vector3()` or `Dictionary()` etc. This is also like in Python.

The reason of that what it's (imho) more laconic and neat. Especially in case with custom data structures witch can be instanced very often.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Described above.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

For example:
```gdscript
var value: CustomCollection = CustomCollection()
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No. Beacause it's a core syntax.

### Is there a reason why this should be core and not an add-on in the asset library?

Beacause it's a GDScript core improvement

**Comment by Mickeon** [comment]
If GDScript had custom data structure, this could probably be the syntax used for them (Akin to writing `Color()`, `Vector3()`...). But, I do find the explicit `.new()` to be much more readable as the creation of a **new** instance of that specific object. Being able to omit it may be, on the contrary of what the proposal may suggest, inconsistent with GDScript's prebuilt data structures.

**Comment by Calinou** [comment]
> Just like Vector3() or Dictionary() etc.

These are not Object-derived types, so I don't think the comparison holds up. Vector3 is also passed by value, not by reference.

**Comment by filantus** [comment]
@Calinou 
> Code is read way more often than it is written

I'm agree. And that's actually what i mean. I'm not lazy to write a few symbols more, but i like how to look code without `.new()`. Just thinking it's slightly more elegant.

Also, mentioned structs is's very good news! Eager to try it. Thanks. 👍
But what about case with CustomCollection, where susposed to be some custom methods?

**Comment by dalexeev** [comment]
I don't mind this suggestion, but I want to point out that the `my_lambda()` syntax is also not supported for performance reasons (you have to `my_lambda.call()`). Probably, in the case of `MyClass.new()`, this problem is not so significant, because 99% of the time `MyClass` is a global name or a constant, but for consistency I'd rather leave `.new()`.

```gdscript
var script := GDScript.new()
script.source_code = "func run(): print('Hello!')"
script.reload()
var instance = script.new() # Rare case when .new() is used with a non-constant.
instance.run()
```

**Comment by filantus** [comment]
@dalexeev Yeah, `lambda.call()` syntax frustrate me a little bit as well. I'm used to the Python syntax where lambdas and regular functions in that case are practically same. I think for callable object use `.call` is redundant. From a visual point of view. Can't say about performance. To me it's looks like to use `obj.get('some_prop')` syntax everywhere, instead of just `obj.some_prop`. Annoying. But anyway i am very thankful to godot devs for great stuff and colossal work that are they do.

**Comment by PerMalmberg** [comment]
Adding my thumbs up to this one. Being able to create instances of my own GDScript classes using the same syntax as for `Vector3` and other built-in classes would make things more readable and consistent. Using `.new()` feels out of place.
